### PR TITLE
Improve CLI docs

### PR DESCRIPTION
**Pull request by tigerros**
* Fixed all the typos (hopefully).
* Improved the bad English. This encompasses grammar, vocabulary, generally awkward phrases, etc. The result is easier to read and understand, and in some cases shorter.
* Made some formatting improvements. This includes making formatting more consistent, but I changed some of it to be better, at least in my opinion.
* General fixes. For example the `dioxus-cli` crate README points to a non-existent documentation URL. One more URL I recall is the Dioxus logo at the [CLI intro page](https://dioxuslabs.com/learn/0.4/CLI), which leads to nowhere. Removed it because I don't think it's necessary, and no other chapter has it.
* Removed certain unnecessities, the biggest one being the commands documentation. I think this is unnecessary because of the `dx --help` command. It basically just repeats what `dx --help` says and it's an extra burden since it needs to be updated every time a command gets added/updated/removed.
* Improved plugin documentation. I added a warning for the reader to ignore all of it, which is ironic, but the fact is that plugins are probably going to change. But by the time I learned that, I already made some changes to the plugin docs. This includes a better guide on how to get started (although still bad since the plugin system is too messy), and general changes like the aforementioned English or formatting.
* And more. I can't list it all, but I didn't do any destructive changes. The content might be reformatted (to be shorter or easier to understand), but the meaning isn't lost.

### CODE REVIEW

File: packages/cli/docs/src/creating.md [hunk 1]
```diff
@@ -1,39 +1,37 @@
...
-First, run the `dx create` command to create a new project ready to be used with Dioxus and the Dioxus CLI:
-
+First, run the `dx create` command to create a new project:
 ```
 dx create hello-dioxus
 ```
 
-> It will clone a default template from github template: [DioxusLabs/dioxus-template](https://github.com/DioxusLabs/dioxus-template)
+> It will clone this [template](https://github.com/DioxusLabs/dioxus-template).
 > This default template is use for `web` platform application.
```
Comment (esimkowitz) [inline packages/cli/docs/src/creating.md:13]:
```suggestion
> This default template is used for `web` platform application.
```

**Comment by ealmloff** [comment]
This looks great! I agree that removing command documentation makes sense. The published CLI docs now live at https://github.com/DioxusLabs/docsite/tree/master/docs-src/0.4/en/CLI, not in this repo. See #1408

**Comment by tigerros** [comment]
> The published CLI docs now live at https://github.com/DioxusLabs/docsite/tree/master/docs-src/0.4/en/CLI, not in this repo. See https://github.com/DioxusLabs/dioxus/pull/1408

Really? `docs-src/0.4/SUMMARY.md` has incorrect paths. It ignores the fact that everything is in the `docs-rs/0.4/en` directory. Does it still work? Snippet from `SUMMARY.md`:
```md
# Summary

[Introduction](index.md)
```
File structure:
```
0.4/
  ...
  SUMMARY.md
  en/
    index.md
    ...
```

**Comment by tigerros** [comment]
Okay. I'll make a PR in the docsite repo, but this PR also changes the `dioxus-cli` crate README, since it also had some issues, and that README is not in the docsite repo. So I think this should be merged, but perhaps without the docs directory.

**Review by ealmloff — APPROVED** [review]

**Pull request merged**